bam_mate.c

```diff
@@ -155,9 +155,27 @@ static bool plausibly_properly_paired(bam1_t* a, bam1_t* b)
         return false;
 }
 
-static void sync_mq(bam1_t* src, bam1_t* dest)
+static void bam_format_cigar(const bam1_t* b, kstring_t* str)
+{
+    // An empty cigar is a special case return "*" rather than ""
+    if (b->core.n_cigar == 0) {
+        kputc('*', str);
+        return;
+    }
+
+    const uint32_t *cigar = bam_get_cigar(b);
+    uint16_t i;
```

Comment:
This should be uint32_t as although uint16_t is sufficient for current BAMs, we're already bumping into this limit (and CRAM can avoid the 16-bit size limit).